File: frontend/src/api/ai.ts
import { useMutation, useQuery } from "@tanstack/react-query";

import { api } from "@/lib/apiClient";

export type AiStatus = {
  engine: string;
  sidecar_reachable: boolean;
  sidecar_mode: string | null;
  gpu_available: boolean | null;
  vila_loaded: boolean | null;
};

export type AskResponse = {
  answer: string;
  engine: string;
  mode: string | null;
};

export function useAiStatus() {
  return useQuery({
    queryKey: ["ai", "status"],
    queryFn: async () => (await api.get<AiStatus>("/ai/status")).data,
    staleTime: 60_000,
  });
}

export function useAskStudy(studyId: string) {
  return useMutation({
    mutationFn: async (question: string) =>
      (await api.post<AskResponse>(`/studies/${studyId}/ask`, { question })).data,
  });
}

File: backend/app/schemas/ai.py
"""AI assistant schemas (VILA-M3)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class AiStatusOut(BaseModel):
    engine: str
    sidecar_reachable: bool
    sidecar_mode: str | None = None
    gpu_available: bool | None = None
    vila_loaded: bool | None = None


class AskRequest(BaseModel):
    question: str = Field(..., min_length=1, max_length=4000)


class AskResponse(BaseModel):
    answer: str
    engine: str
    mode: str | None = None
